insights/client/__init__.py

```diff
@@ -209,6 +224,25 @@ def verify(self, egg_path, gpg_key=constants.pub_gpg_path):
                     'stdout': None,
                     'rc': 0}
 
+        # check if the provided files (egg and gpg) actually exist
+        if egg_path and not os.path.isfile(egg_path):
```

Comment:
Perhaps move this `if` block above the `gpg` option check to make sure it's done even when gpg verification is disabled.